chore: add comprehensive Go README (#417)
iotaledger/iota-rust-sdk: bindings/go/README.md

# IOTA SDK - Go Bindings

Go bindings for the IOTA SDK, enabling Go developers to interact with the IOTA blockchain.

## Installation

To use the IOTA SDK in your Go project, add it as a dependency:

```bash
go get github.com/iotaledger/iota-sdk-go
```

The SDK requires the native IOTA SDK FFI library to be installed on your system. Pre-built libraries are available for:
- macOS (x86_64 and ARM64)
- Linux (x86_64 and ARM64)

> - Windows (x86_64 and ARM64)

Reviewer: @copilot this is not true as we are pushing the libraries to github.com/iotaledger/iota-sdk-go

## Quick Start

Here's a simple example that queries the chain ID from the IOTA network:

```go
package main

import (
	"fmt"
	"log"

	"github.com/iotaledger/iota-sdk-go/iota_sdk"
)

func main() {
	// Create a GraphQL client connected to devnet
	client := iota_sdk.GraphQlClientNewDevnet()

	// Query the chain ID
	chainID, err := client.ChainId()
	if err.(*iota_sdk.SdkFfiError) != nil {
		log.Fatalf("Failed to get chain ID: %v", err)
	}
	fmt.Println("Chain ID:", chainID)
}
```

## Usage

The SDK provides GraphQL client functionality to interact with IOTA:

```go
// Connect to devnet
client := iota_sdk.GraphQlClientNewDevnet()

// Connect to testnet
client := iota_sdk.GraphQlClientNewTestnet()

// Connect to mainnet
client := iota_sdk.GraphQlClientNewMainnet()

// Connect to a custom endpoint
client := iota_sdk.GraphQlClientNew("https://your-endpoint.com")
```

## Examples

> More examples are available in the [examples directory](examples/), including:

Reviewer: @copilot Please out the complete examples URL as this readme is meant to be used in github.com/iotaledger/iota-sdk-go as well and the relative link would be broken

# IOTA SDK - Go Bindings

> Go bindings for the IOTA SDK, enabling Go developers to interact with the IOTA blockchain.

Reviewer: ```suggestion
Go bindings for the IOTA SDK, enabling Go developers to interact with the IOTA network.
```

## Installation

To use the IOTA SDK in your Go project, add it as a dependency:

```bash
go get github.com/iotaledger/iota-sdk-go
```

The package includes pre-built native libraries for:

- macOS (x86_64 and ARM64)
- Linux (x86_64 and ARM64)
- Windows (x86_64 and ARM64)

## Quick Start

> Here's a simple example that queries the chain ID from the IOTA network:

Reviewer: ```suggestion
Here is a simple example that queries the chain ID from the IOTA network:
```

Verify by running `make --version`, `go version`, and `uniffi-bindgen-go --version`.

### Generate Go bindings

```bash
make go
```

### Run Go examples

```sh
make go-example chain_id
```

## License

> Apache-2.0

Reviewer: ```suggestion
```
